src/Controllers/DatabaseController.js

```diff
@@ -1207,6 +1207,9 @@ class DatabaseController {
                 `Invalid field name: ${fieldName}.`
               );
             }
+            if (!schema.fields[fieldName.split('.')[0]]) {
```

Comment:
What should be the expected outcome when sorting by non-existing field? Ignoring the field or throwing a 400 "invalid field name" error?

Reply:
I think it should just delete that sort key as a safe failure, but we can throw here if you think it's a better approach

Reply:
How does Parse Server behave in similar situations, e.g. when using a non-existing field in an equalTo condition?

Reply:
It succeeds and returns nothing, e.g:

```js
const query = new Parse.Query('a').equalTo('b','c');
const obj = await query.first()
// obj is undefined
```

Reply:
Interesting, in which cases is the error "invalid key name" used then?

Reply:
I think that's mostly used to protect internal / sensitive keys (prevent sorting by authdata for example)

Reply:
Well, if query conditions fail gracefully, then I think sort should do the same. So we can keep the approach as is.